### PR TITLE
Update development tooling

### PR DESCRIPTION
**Pull request by lemonmade**
This PR updates the development tooling for the UI extension repo:

- Moves off the (very old) `@sewing-kit` dependencies to Shopify's `loom` project
- Updated TypeScript, Prettier, and ESLint to their latest versions
- Updated the CI workflows to be a bit easier to grok

There should not be any actual changes to the packages' functionality from this change, so I will not be bumping any dependencies after merging this. Sorry for the large amount of changes, they are pretty much exclusively Prettier's doing :/

**Update:** actually, I think I will have to publish a version to avoid other folks unnecessarily getting these bumps in  a future version change. I also had to do some changes to the `types`/ `typeVersions` field in every package's `package.json` to match how Loom expects you to declare types for a package. I will do a patch release for each package.

### CODE REVIEW

File: packages/admin-ui-extensions/package.json [hunk 1]
```diff
@@ -4,20 +4,33 @@
   "main": "index.js",
   "module": "index.mjs",
   "esnext": "index.esnext",
-  "types": "index.d.ts",
+  "types": "./build/ts/index.d.ts",
+  "typesVersions": {
```
Comment (vividviolet) [inline packages/admin-ui-extensions/package.json:8]: Can you clarify why we need to declare these again here?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 node_modules
+.loom
 .sewing-kit
```
Comment (vividviolet) [inline .gitignore:3]: Can we eventually get rid of this?

**Review by vividviolet — APPROVED** [review]
Tested the admin-ui-extensions-react library - everything works. Just need to fix the conflicts on the md files

File: .gitignore (hunk 2 above)
Comment (lemonmade) [reply .gitignore:3]: We can, maybe... I left it so people who have the repo already do not suddenly have a bunch of formally-gitignored files that now show up as added.

**Pull request merged**